flatgfa/src/cmds.rs

```diff
@@ -318,3 +320,199 @@ pub fn depth(gfa: &flatgfa::FlatGFA) {
         );
     }
 }
+
+
+/// chop the segments in a graph into sizes of N or smaller
+#[derive(FromArgs, PartialEq, Debug)]
+#[argh(subcommand, name = "chop")]
+pub struct Chop {
+    /// maximimum segment size.
+    // Use c in keeping with odgi convention
+    #[argh(option, short = 'c')]
+    c: usize,
+
+    /// compute new links
+    #[argh(switch, short = 'l')]
+    l: bool,
+}
+
+/// Chop a graph into segments of size no larger than c
+/// By default, compact node ids
+/// CIGAR strings, links, and optional Segment data are invalidated by chop
+/// Generates a new graph, rather than modifying the old one in place
+pub fn chop<'a>(
+    gfa: &'a flatgfa::FlatGFA<'a>,
+    args: Chop,
+) -> Result<flatgfa::HeapGFAStore, &'static str> {
+
+    let mut flat = flatgfa::HeapGFAStore::default();        
+
+    let mut seg_map: Vec<(Id<Segment>, Id<Segment>)> = Vec::new();
+    let mut max_node_id = 1;
+
+    fn empty_span<T>() -> Span<T> {
+        Span::new(Id::new(0), Id::new(0))
+    }
```

Comment:
I think you have a utility method for this now (`Span::new_empty()`)?

Reply:
I have replaced this and the other calls to `empty_span` with the utility method